### DISCUSSION TITLE
tasmota32-zbbridge Zigbee2Tasmota (Z2T) builds for ESP32?

### DISCUSSION THREAD

**Discussion by Hedda** [arendst/Tasmota#11913] [Q&A] [answered]
Are there now also "zbbridge" specific build versions of Tasmota32 that should be used on ESP32 based Zigbee bridges/gateways?

That is, Tasmota for ESP32 release builds with Zigbee2Tasmota (Z2T) enabled by default?

https://tasmota.github.io/docs/Zigbee/

There look to be "tasmota-zbbridge.bin" binaries for ESP8266 and ESP8285 but no "tasmota32-zbbridge.bin" for ESP32 as of yet?

https://github.com/arendst/Tasmota/releases

Reason I ask is that s-hadinger wrote "**_The Tube's cc2652p2 is now officially supported. The EFR32 will be shortly._**" in https://github.com/arendst/Tasmota/issues/9081

It refers to "Tube's Zigbee Gateways" which are ESP32 ([Wireless-Tag WT32-ETH01](http://www.wireless-tag.com/portfolio/wt32-eth01/)) open-source hardware Zigbee bridges by @tube0013

https://github.com/tube0013/tube_gateways

Those are ESP32 based bridges with the latest Silcon Labs EFR32 radios running EmberZNet Zigbee stack (EZSP v8 API / serial protocol).

PS: tube makes and sells versions with Texas Instruments CC2652P/CC2652P2 or Silabs EFR32 (both EFR32MG12 and EFR32MG21 variants):

https://www.tubeszb.com/shop/coordinators/2

**Comment by barbudor** [comment]
tagging @s-hadinger

**Comment by s-hadinger** [comment]
It's only a question of critical mass to add pre-compiled binaries. Sonoff ZBBridge is mainstream and it makes sense to have pre-compiled binaries. ESP32 version are still more pricy and not so much deployed.

That would also require 2 versions, one for ZNP (TI CC2652 chip) and one for EZSP (EFR32 chip).

**Comment by Hedda** [comment, marked as answer]
While not mainstream you can also make cheap DIY version by ripping IKEA TRÅDFRI ICC-A-1 module out of IKEA Trådfri products:

https://github.com/MattWestb/IKEA-TRADFRI-ICC-A-1-Module

https://github.com/basilfx/TRADFRI-Hacking

Need IKEA TRÅDFRI ICC-A-1 module, ESP32 module like Wemos Mini D1 ESP32 or Wireless-Tag WT32-ETH01, and a power-supply.

If you base it on the IKEA Trådfri Remote Control then you do not even relly have to rip out the IKEA TRÅDFRI ICC-A-1 module, see:

https://www.heise.de/make/artikel/Ikea-Tradfri-Anleitung-fuer-ein-ESP8266-Lampen-Gateway-3598411.html

**Reply by barbudor** [reply]
If you have the skills to assemble your own ESP32 Zigbee bridge, you should also be able to build your own binary.

So is the question about providing pre-built binaries ?
Or is it not working ?

If some more doc is required, you can still propose doc updates to make it easier like providing ready `user_config_override.h` and ready `platformio_tasmota_cenv.ini` entries

**Reply by Hedda** [reply]
> So is the question about providing pre-built binaries ?
> Or is it not working ?

My question was only about providing pre-built binaries at https://github.com/arendst/Tasmota/releases

**Reply by sfromis** [reply]
If it in the future becomes a commonly used combo, it would become a possibility to provide a precompiled binary for that config.

**Comment by s-hadinger** [comment]
I'm happy to announce that Tube's Zigbee Coordinator based on EFR32 is now also supported. The doc has been updated.

Just be aware that you need to downgrade EZSP to 6.7.9. The firmware is provided and it is simply done via Tasmota through the integrated XModem client.

**Reply by tube0013** [reply]
Thanks!

**Comment by Hedda** [comment]
> If you have the skills to assemble your own ESP32 Zigbee bridge, you should also be able to build your own binary.

FYI, putting together your own inexpensive DIY  Zigbee gateway/bridge hardware is actually not complicated at all today.

TI and Silabs Zigbee modules available use serial communication so once flashed it only need to connect power + serial TX and RX.

Again, can either buy a Zigbee SMD module (RF serial module) or re-flash such a module from an IKEA "Trådfri" device.

The recommendation is then to use an ESP32 board with a (wired) Ethernet port like the [Wireless-Tag WT32-ETH01](http://www.wireless-tag.com/portfolio/wt32-eth01/) or [LILYGO TTGO T-Internet-POE](http://www.lilygo.cn/prod_view.aspx?TypeId=50033&Id=1307), (as using WiFi does not always make for a reliable serial-to-IP bridge, at least not for the Silabs EZSP serial protocol).

Links below for reference:

- DIY Zigbee gateway/bridge device example based on IKEA TRÅDFRI ICC-A-1 Modules (Silicon Labs EFR32MG12 or EFR32MG21):
  - https://github.com/MattWestb/IKEA-TRADFRI-ICC-A-1-Module/tree/master/Tasmota
  - https://github.com/MattWestb/IKEA-TRADFRI-ICC-A-1-Module

- DIY Zigbee gateway/bridge device example based on Ebyte E180-ZG120B or E180-ZG120B-TB (Silicon Labs EFR32MG12):
  - https://github.com/zigpy/zigpy/discussions/584

-  Building a Texas Instruments CC2530 based DIY Zigbee gateway/bridge is very much similar to making a DIY CC2530 router:
   - https://www.zigbee2mqtt.io/how_tos/how_to_create_a_cc2530_router.html

**Reply by Hedda** [reply]
> Again, can either buy a Zigbee SMD module (RF serial module) or re-flash such a module from an IKEA "Trådfri" device.

FYI, @MattWestb is also experimenting with reflashing Zigbee modules used inside newer Tuya and Lidl Silvercrest devices:

https://github.com/zigpy/zigpy/discussions/738

https://github.com/MattWestb/EFR32-FW

Apparently, the new Tuya ZS3L and rebranded LED strips also contain a similar EFR32MG21 based PCB package SMD module:

![image](https://user-images.githubusercontent.com/6320001/118454486-b634f280-b6f8-11eb-974d-8a28a32cf734.png)

**Reply by MattWestb** [reply]
Both IKEA ICC-A-1 and the tuya ZS3L with one standard EZSP 6.7.X is ruing well with on standard tasmota with Zigbee bridge on one WeMos D1 Mini and have doing for over one  year and dont need any extra for working.

Only tricky is flashing new bootloader then both have OTA bootloader that we cant using.

And the most easy way for getting one (or 2) MG21 NCP is baying on family pack with then new E27 CWS3 and E2001 remote and the remotes is not locked and can being re flashed if like going back to "remote mode".

![IMG_20210505_143614](https://user-images.githubusercontent.com/49618193/118459767-593b3b80-b6fc-11eb-9671-5ca17fae91d5.jpg)

E2001 Hocked up and dumped all the flash :-))

**Reply by Hedda** [reply]
FYI, DIY bridge can also use board from ITead's new “ZBDongle-E” Zigbee dongle based on EFR32MG21 SoC from Silicon Labs, see:

https://itead.cc/product/zigbee-3-0-usb-dongle/

Here is the link to FCC filing/report for the new “ZBDongle-E” (a.k.a. "ZBD-E" ) adapter which contains copy of the user manual + a lot more detailed + many more internal and external pictures:

[https://fccid.io/2APN5ZBD-E](https://fccid.io/2APN5ZBD-E)

[https://fccid.io/2APN5ZBD-E/User-Manual/User-Manual-6012461.pdf](https://fccid.io/2APN5ZBD-E/User-Manual/User-Manual-6012461.pdf)

[https://fccid.io/2APN5ZBD-E/Internal-Photos/Internal-Photos-6012455](https://fccid.io/2APN5ZBD-E/Internal-Photos/Internal-Photos-6012455)

[https://fccid.io/2APN5ZBD-E/External-Photos/External-Photos-6012451](https://fccid.io/2APN5ZBD-E/External-Photos/External-Photos-6012451)

It uses the same [EFR32MG21A020F768IM32 (EFR32 Mighty Gecko Series 2) SoC](https://www.silabs.com/wireless/zigbee/efr32mg21-series-2-socs/device.efr32mg21a020f768im32?tab=specs) used on the first ITead Zigbee 3.0 USB Dongle, as well as on CoolKit Technologies "SM-011 V1.0" module used in th ITead Sonoff ZBBridge Zigbee Bridge and in Tuya ZS3L and Tuya ZSLC5 SoC modules. It includes a 80 MHz ARM Cortex-M33 core, 768 Flash Storage, 64 RAM, in a QFN32 package, which radio has + 20 dBm maximum power output and receive sensitivity of -104 (250 kbps O-QPSK DSSS) dBm. 

![image](https://user-images.githubusercontent.com/6320001/182600362-d76008a4-b380-4504-9551-f7cca240bf90.png)

![image](https://user-images.githubusercontent.com/6320001/182599930-da55b43d-438e-42d6-9adf-1230188dd567.png)

![image](https://user-images.githubusercontent.com/6320001/182600312-8e8fe387-9457-4c5a-a4e9-a6fea1afe78c.png)

![image](https://user-images.githubusercontent.com/6320001/182401298-a5477c23-6be3-4071-aeaf-e473091264df.png)

There is further details and more users discussion here:

https://github.com/Koenkk/zigbee2mqtt/discussions/13373

By the way, the same Silabs EFR32MG21 SoC will also be used in the official "Home Assistant SkyConnect USB Stick", see:

https://community.home-assistant.io/t/home-assistant-skyconnect-usb-stick-announced-will-be-compatible-with-both-zigbee-and-thread-including-matter-chip-over-thread/433594

You can submit your interest in the official "Home Assistant SkyConnect USB Stick" here:

https://docs.google.com/forms/d/e/1FAIpQLScEjHSBJszUZfgO3MIDO51IHr3Oeohcs8BLpRIjY1liJ58IpA/viewform

PS: These can also support multiprotocol firmware which can allow availability to use Zigbee and Thread protocols at the same time.
